### PR TITLE
JAPI-512 Github action based HPCC/Spark evn for junit tests

### PR DESCRIPTION
**Pull request by jpmcmu**
- Modified Github actions to test against a K8s HPCC cluster
- Modified Spark JUnits to use mvn integration tests
- Modified Spark integration tests to use Datasource API
- Removed outdated tests
- Modified POM file to fix dependency conflicts

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

**Review by rpastrana — CHANGES_REQUESTED** [review]
@jpmcmu looks good, left a few comments

File: .github/workflows/spark-hpcc-remote-test.yml [hunk 1]
```diff
@@ -18,10 +18,12 @@ jobs:
       matrix:
         versions:
           - spark: 3.2.1
-            hadoop: hadoop3.2
+            spark-url: https://archive.apache.org/dist/spark/spark-3.2.1/spark-3.2.1-bin-hadoop3.2.tgz
```
Comment (rpastrana) [inline .github/workflows/spark-hpcc-remote-test.yml:18]: are these pulled every time?

File: .github/workflows/sparkconnectorbuildonmerge.yml [hunk 2]
```diff
@@ -1,20 +0,0 @@
-name: HPCC Spark connector build on merge
```
Comment (rpastrana) [inline .github/workflows/sparkconnectorbuildonmerge.yml:1]: It seems fine to remove this action, but we do need to ensure the build doesn't break upon a merge onto candidate-* and or master. Any reason not to trigger spark-hpcc-remote-test.yml on push?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu looks good, had a couple of questions.

File: .github/workflows/spark-hpcc-remote-test.yml (hunk 1 above)
Comment (jpmcmu) [reply .github/workflows/spark-hpcc-remote-test.yml:18]: Good question, I have a change in the version that includes the changes from HPCCJ-523 that will won't require a redownload. Will push that up after HPCC4J-523 is merged.

**Pull request merged**